### DISCUSSION TITLE
What to do if `transform` is always `null` ?

### DISCUSSION THREAD

**Discussion by OwenMelbz** [clauderic/dnd-kit#357] [Q&A]
When calling:

```js
const {
        attributes,
        listeners,
        setNodeRef,
        transform,
        transition,
    } = useSortable({
        id: props.id,
    })
```

I get back `transform` however it is always `null` even whilst dragging it's coming back empty.

It also means that none of the re-ordering animations are working.

Are there an FAQs or common cases that this wouldn't work?

We're just rendering a list of divs, some are flexed but nothing like fixed, or virtual lists or anything :(

Thanks

**Comment by clauderic** [comment]
Hey @OwenMelbz, kind of difficult to help you diagnose the issue without a reproduction case of the issue. 

Assuming you've set everything up properly, in the past others have reported these types of issues when having multiple versions of `@dnd-kit/core` installed locally. Double check your `package-lock.json` or `yarn.lock` file to make sure you only have a single version installed.

**Reply by theMostCuriousHomunculus** [reply]
hey i think i know what is causing this!  i just spent the last day working through a similar issue.  i suspect that the array of items you are trying to reorder, @OwenMelbz , is not an array of primitive types (strings, numbers, etc...), but rather an array of objects.  the issue i had was that i overlooked this line in the documentation:

"In addition to the DndContext provider,  the Sortable preset requires its own context provider that contains the sorted array of the **unique identifiers** associated to each sortable item" (emphasis added)

in other words, you don't pass items={items} into your sortable context; you pass something like items={items.map(item => item.id)} (or whatever field on the object you are using as your unique identifier in your sortable item component)

@clauderic , this took me quite a while to catch (now everything is working beautifully 😁).  all the examples i found online using this library use arrays of primitive types; i think it would be useful to include an example sorting an array of objects, or perhaps include a common pitfall callout stressing that the items prop of the sortable container needs to contain ONLY the unique identifiers.

hope this is your issue too, @OwenMelbz !  and thanks for the wonderful library, @clauderic !

**Reply by rymanso** [reply]
Thank you @theMostCuriousHomunculus for coming back to this thread to post this. you just saved me a day!

**Reply by sanli** [reply]
Thank you @theMostCuriousHomunculus, I spend one day to dig this problem.

**Reply by nairolf81** [reply]
Many thanks for the hint, I created an example with objects based upon fork
https://codesandbox.io/s/330g6o

**Reply by Werthis** [reply]
thanks @theMostCuriousHomunculus and @nairolf81

**Reply by usman-alfain** [reply]
Ahh! thanks man. succuessfully wasted a week on this and now got its solution.
Thanks..

**Reply by ishar19** [reply]
Thanks @theMostCuriousHomunculus , was stuck on this for 2 days. you just saved me a ton.

**Reply by Ahmed1995mansour** [reply]
Thanks @theMostCuriousHomunculus, You saved me, man.

**Reply by jmlord** [reply]
Just a thought: if so many of us get caught by this pitfall, maybe the library should fail explicitly when provided with an object list?

**Reply by yardenroee** [reply]
> Many thanks for the hint, I created an example with objects based upon fork https://codesandbox.io/s/330g6o

I just wanted to let you know that I love you

**Reply by kendidang2k** [reply]
Thank u so much, It's actually work.

**Reply by jgabriel98** [reply]
~man you just saved my whole week~
never mind, still getting `transform` as `null` when changing the inner content of draggable item

**Reply by tomasmezdej** [reply]
@theMostCuriousHomunculus many thanks

**Reply by SummerYch** [reply]
@theMostCuriousHomunculus OMG！！！！！！！！its worked!!!!!!!!!!!!!!!!!THANK YOU!!!!!!!!!!!!!!!!!!!!

**Reply by escline** [reply]
2 days wasted here as well. Had me pulling my hair out, as I had a working solution in another part of my app. Glad I came across this discussion. I agree that this needs a huge callout in the docs. Or at the very least, a thrown warning if the SortableContext items prop is not an array of primitive types.

**Comment by GNUGradyn** [comment]
Anyone know what to do if this happens with only 1 dnd-kit installed? Transform and transition are always null for me
![image](https://github.com/clauderic/dnd-kit/assets/20762604/3abe312d-9581-44e9-a7fe-3cbfcde5a50d)

**Reply by GNUGradyn** [reply]
I made a whole bug report on this with a video and a MRE https://github.com/clauderic/dnd-kit/issues/1136
